File: packages/browser/browser/factory.py
from __future__ import annotations

import os
from typing import Literal

from browser.agent_browser_backend import AgentBrowserBackend
from browser.browser_use_backend import BrowserUseBackend
from browser.errors import BrowserActionError
from browser.playwright_backend import PlaywrightBackend
from browser.types import BrowserBackend

BrowserBackendName = Literal["agent-browser", "browser-use", "playwright"]


def create_browser_backend(backend: BrowserBackendName | None = None) -> BrowserBackend:
    selected = backend or os.environ.get("MANTLE_BROWSER_BACKEND", "agent-browser")
    if selected == "agent-browser":
        return AgentBrowserBackend()
    if selected == "browser-use":
        return BrowserUseBackend()
    if selected == "playwright":
        return PlaywrightBackend()
    raise BrowserActionError(f"Unsupported browser backend: {selected}")

File: packages/browser/src/factory.ts
import { AgentBrowserBackend } from "./agent-browser-backend.js";
import { BrowserUseBackend } from "./browser-use-backend.js";
import type { BrowserBackend } from "./types.js";

export type BrowserBackendName = "agent-browser" | "browser-use";

export function createBrowserBackend(backend?: BrowserBackendName): BrowserBackend {
  const selected = backend ?? process.env.MANTLE_BROWSER_BACKEND ?? "agent-browser";
  if (selected === "agent-browser") {
    return new AgentBrowserBackend();
  }
  if (selected === "browser-use") {
    return new BrowserUseBackend();
  }
  throw new Error(`Unsupported browser backend: ${selected}`);
}
